### DISCUSSION TITLE
Mux on mux (custom board) - no midi above analog control 63

### DISCUSSION THREAD

**Discussion by eurorackworld** [shanteacontrols/OpenDeck#203] [Q&A] [answered]
I have a question on an issue, I'll try to post some details on the configuration to make a bit clear what setup I have.

I understand that this is hard to judge based on vague photos and no schematics. But maybe someone can push me in the right direction of things to test/check. It feels like it has to do with selecting S1, S2 and S3 , as those are needed to move to the next set of inputs in the shift register if I understand correctly. But those pins are also connected and configured OK it seems.
- I will try another firmware creation and upload, but that would be in principle the same config, so no big changes could come from that.
- I will try another RPI as well, but never had a damaged one before, would be weird if for this one suddenly those pins fails.

**The issue:**
- When I connected boards with potentiometers to the setup, only the first controls up to 63 (so ID 0 to ID63) are working, everything above 63 doesn't work. 

**The setup:**
- RPI Pico
- 5 boards each with 3 hc4067 multiplexers, in total 15 multiplexers going to 1 global hc4067 mux (so everything is 16 channels)
- Open mux input aren't connected to ground
- In opendeck I see 255 analog controls mentioned
- There is a control board which connects to all 5 sub boards
- Every connector to a board of course enables 3 mux signals, which are routed to the global mux

**Already did**
- Triple checked if all boards work (they do), if all connection are correct on the PCB's (I checked all Sx connections, from the general mux to the RPI, but also all mux's to the general mux. All connections are fine.
- All boards work fine if I just put them on the first 0-63 controls, I tried if for example control 80 or 100 did work, also not. So really on the first 0-63 ones. 
- Switching on/off various controls to see if it would show something strange, all fine.
- **Important to know**: the issue isn't started with the 2nd board/connector, as 64 controls are working (4x mux) so 3 mux from the 1st board and 1 from the next board. I don't think it is a problem of the connections on the PCB itself, they all are triple checked before pcb creation and also on the final board itself.

**Config file:**
<img width="206" alt="image" src="https://user-images.githubusercontent.com/29734704/236539387-5581d81f-c12f-47a7-be72-04dcf193542a.png">

**Total controller**
IMAGES REMOVED

**Per board**
IMAGES REMOVED
I use these boards normally with 3 mux on a RPI, but I've made a custom mini PCB which features a connector to connect all important signals to the control board
Ignore the green wires, those are due to missing routes in the PCB, luckily I already have new ones which don't require those 4 green bypasses anymore :P

**Control board**
IMAGES REMOVED
Hopefully the labelling helps a bit on understanding what I did here.

**Comment by paradajz** [comment, marked as answer]
This was a fairly stupid bug:
https://github.com/shanteacontrols/OpenDeck/commit/64633d3cff5ce0c425925934110f4d93a11c67ed

If you pull the latest changes and build the firmware against the latest commit it should work fine.

**Reply by eurorackworld** [reply]
A small step for you but a giant leap for me, that fix just instantly gave me 180 potentiometers sending MIDI !!

I'm crying both of joy (it works) and pain (I checked so many things, happy to see my boards work perfect).

Now continuing with the buttons and encoders (or leds, not sure yet what I want..) and then it's ready!

Massive thanks again for replying so fast, made my day!

**Reply by paradajz** [reply]
Hey glad it works out :) Make sure to post your ginormous controller in the show-and-tell section here ;)

**Comment by stllvdv** [comment]
> * 5 boards each with 3 hc4067 multiplexers, in total 15 multiplexers going to 1 global hc4067 mux (so everything is 16 channels)
> * There is a control board which connects to all 5 sub boards
> * Every connector to a board of course enables 3 mux signals, which are routed to the global mux

@eurorackworld: Would you be willing to share schematics for this? Would love to take a look.

**Reply by eurorackworld** [reply]
Hey sure! I'm currently on holiday so if you can remind me next week then I'll send you whatever you'd like to have, happy to dump some stuff into a zipfile towards you

**Reply by stllvdv** [reply]
Please do! Curious about those 5 (?) boards and how everything is connected… I was under the impression that the MUX ON MUX configuration needs one PICO with multiple MUXES, one MUX being the controller and the rest being nodes.
